ICE with builtin-dynamic-object-size-0.c

Hit this when running `make check`:

```
Starting program: /usr/libexec/gcc/x86_64-pc-linux-gnu/15/cc1 -quiet -iplugindir=/usr/lib/gcc/x86_64-pc-linux-gnu/15/plugin builtin-dynamic-object-size-0.c -iplugindir=/usr/lib/gcc/x86_64-pc-linux-gnu/15/plugin -quiet -dumpbase builtin-dynamic-object-size-0.c -dumpbase-ext .c -mtune=generic -march=x86-64 -O2 -fplugin=../fmetrics.so -fplugin-arg-fmetrics-project=fmetrics -o /dev/null
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib64/libthread_db.so.1".
builtin-dynamic-object-size-0.c: In function ‘main’:
builtin-dynamic-object-size-0.c:646:7: warning: ‘test_parmsz_scaled’ accessing 176 bytes in a region of size 168 [-Wstringop-overflow=]
  646 |   if (test_parmsz_scaled (arr, 44) != 44 * sizeof (int)) /* { dg-warning "-Wstringop-overflow=" } */
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
builtin-dynamic-object-size-0.c:411:1: note: in a call to function ‘test_parmsz_scaled’ declared with attribute ‘access (read_write, 1, 2)’
  411 | test_parmsz_scaled (int *obj, size_t sz)
      | ^~~~~~~~~~~~~~~~~~

Breakpoint 3, internal_error (gmsgid=gmsgid@entry=0x555557fb11f7 "in %s, at %s:%d") at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/diagnostic-global-context.cc:512
512     {
```

```
(gdb) bt
#0  internal_error (gmsgid=gmsgid@entry=0x555557fb11f7 "in %s, at %s:%d") at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/diagnostic-global-context.cc:512
#1  0x0000555556d86b7e in fancy_abort (file=0x555557fbb6a8 "/var/tmp/portage/sys-devel/gcc-15.0.9999/work/gcc-15.0.9999/gcc/passes.cc", line=2150,
    function=0x555557f0d3cb "execute_todo") at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/diagnostic.cc:1657
#2  0x00005555555d0c9a in execute_todo (flags=64) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2150
#3  0x00005555555cf43a in execute_one_pass (pass=warning: RTTI symbol not found for class 'pass_fmetrics'
warning: RTTI symbol not found for class 'pass_fmetrics'
<opt_pass* 0x55555880a4c0 "fmetrics"(369)>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2705
#4  0x0000555557070e4c in execute_pass_list_1 (pass=warning: RTTI symbol not found for class 'pass_fmetrics'
warning: RTTI symbol not found for class 'pass_fmetrics'
<opt_pass* 0x55555880a4c0 "fmetrics"(369)>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2768
#5  0x0000555557070e6b in execute_pass_list_1 (pass=<opt_pass* 0x555558829b40 "*all_optimizations"(-1)>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2769
#6  0x0000555557070bee in execute_pass_list (fn=0x7ffff6c050c0, pass=<optimized out>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2779
#7  0x0000555556fe1626 in cgraph_node::expand (this=<cgraph_node * const 0x7ffff6c08000 "test_strdup"/43>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/context.h:48
#8  0x0000555556f60ed6 in expand_all_functions () at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/cgraphunit.cc:2028
#9  symbol_table::compile (this=0x7ffff7206000) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/cgraphunit.cc:2404
#10 0x00005555577915ee in symbol_table::finalize_compilation_unit (this=0x7ffff7206000) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/cgraphunit.cc:2589
#11 0x00005555577328e1 in compile_file () at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/toplev.cc:478
#12 0x00005555576e222f in do_compile () at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/toplev.cc:2213
#13 toplev::main (this=this@entry=0x7fffffffd566, argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/toplev.cc:2373
#14 0x00005555576e12fc in main (argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/main.cc:39
```

```
(gdb) frame 2
#2  0x00005555555d0c9a in execute_todo (flags=64) at /usr/src/debug/sys-devel/gcc-15.0.9999/gcc-15.0.9999/gcc/passes.cc:2150
2150        gcc_assert (flags & TODO_update_ssa_any);
```

Needs checking to hit it, I guess.

--- a/fmetrics.cc
+++ b/fmetrics.cc
@@ -6,6 +6,7 @@
 #include <builtins.h>
 #include <context.h>
 #include <tree.h>
+#include <tree-cfgcleanup.h>
 #include <tree-object-size.h>
 #include <tree-pass.h>
 #include <tree-pretty-print.h>
@@ -107,6 +108,8 @@ unsigned int
 pass_fmetrics :: execute (function *fun)
 {
   basic_block bb;
+  unsigned int todo = 0;
+
   FOR_EACH_BB_FN (bb, fun)
     {
       gimple_stmt_iterator i;
@@ -146,5 +149,8 @@ pass_fmetrics :: execute (function *fun)
 
   fini_object_sizes ();
 
-  return 0;
+  if (cleanup_tree_cfg (TODO_update_ssa_only_virtuals))
+    todo |= TODO_update_ssa_only_virtuals;
+
+  return todo;
 }

